### PR TITLE
Allow to pass arguments to rustdoc-gui tool

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Very convenient for testing. This is another part of https://github.com/rust-lang/rust/pull/86293

cc @jsha 
r? @Mark-Simulacrum

### CODE REVIEW

**Comment by jsha** [comment]
Looks good to me. Waiting on giving the r+ since you assigned the review to @Mark-Simulacrum

File: src/bootstrap/test.rs [hunk 1]
```diff
@@ -870,6 +870,13 @@ impl Step for RustdocGUI {
             .arg(out_dir)
             .arg("--tests-folder")
             .arg(builder.build.src.join("src/test/rustdoc-gui"));
+        for test_arg in builder.config.cmd.test_args() {
+            if ["--debug", "--no-headless", "--show-text"].iter().any(|a| *a == test_arg) {
+                command.arg(test_arg);
+            } else {
+                panic!("Unknown argument: `{}`", test_arg);
```
Comment (Mark-Simulacrum) [inline src/bootstrap/test.rs:877]: We generally don't filter arguments in *bootstrap* -- why is this logic not in the script being executed?

Comment (GuillaumeGomez) [reply src/bootstrap/test.rs:877]: It is as well, it's true that it's duplicated though... Removing it.

**Comment by GuillaumeGomez** [comment]
@Mark-Simulacrum Removed the duplicated check.

**Comment by GuillaumeGomez** [comment]
Removed the invalid submodule update...

**Pull request merged**